core/src/autogluon/core/scheduler/managers.py

```diff
@@ -0,0 +1,53 @@
+import multiprocessing as mp
+
+from autogluon.core.scheduler.remote import RemoteManager
+from autogluon.core.scheduler.resource import DistributedResourceManager
+
+
+class TaskManagers(object):
+    LOCK = mp.RLock()
```

Comment:
Difference between RLock and Lock?

Reply:
RLock == reentrant lock same thread can re-enter the lock if needed.